### PR TITLE
Add text statistics next to the cursor position

### LINKED ISSUE #164: Word count feature
It should be desirable to have a "document word count" feature. Furthermore, a status line with line number and a column number will be great

### PR DESCRIPTION
**Pull request by xgouchet**
This adds a word count, alpha numeric character count and all character
count in the bottom right corner, with the cursor line and column
position

Fixes #164

### CODE REVIEW

**Comment by mitya57** [comment]
Sorry for the delay with reviewing your requests.
The idea looks fine, but I would like to make this be:

* separate from the cursor position panel (maybe you can make this have a different background color?);
* translatable (wrapping this in a `tr()` and adding a translator comment would be nice).

Also, did you perform any performance checks? How slow is it on large (or huge) documents?

I can look at these issues myself, but in this case you will need to wait a bit more.

**Comment by xgouchet** [comment]
Hi, good idea about the translation, I'm going to add it. 

I thought about the performance issue, and might add an option to enable / disable this feature. 
I also thought of using some heuristic to guess the wordcount without having to actually parse the whole document (the QEditText already gives the character count, and we can divide this number by an average character per count value).

File: ReText/editor.py [hunk 1]
```diff
@@ -408,15 +438,72 @@ def leaveEvent(self, event):
...
+		palette.setColor(QPalette.Window, colorValues['statsArea'])
+		palette.setColor(QPalette.WindowText,
+						 self.editor.palette().color(QPalette.WindowText))
 		self.setPalette(palette)
 
 	def mousePressEvent(self, event):
 		pos = self.mapToParent(event.pos())
 		pos.setX(pos.x() - self.editor.lineNumberAreaWidth())
 		newEvent = QMouseEvent(event.type(), pos,
-		                       event.button(), event.buttons(),
```
Comment (mitya57) [inline ReText/editor.py]: This file uses tabs for indentation, but spaces for alignment. Please do not change this.

File: ReText/editor.py [hunk 2]
```diff
@@ -334,6 +342,28 @@ def installFakeVimHandler(self):
...
+		for c in text:
+			characterCount += 1
+			isWordCharacter = c.isalnum()
+			if isWordCharacter:
+				alphaNumCount += 1
+			if wasWordCharacter and not isWordCharacter:
+				wordCount += 1
+			wasWordCharacter = isWordCharacter
+		end = time.clock()
+		value = int((end - start) * 1000)
```
Comment (mitya57) [inline ReText/editor.py]: Are these start / end / value lines a leftover debug? Can you please share your time measurements?

File: ReText/editor.py [hunk 3]
```diff
@@ -408,15 +438,72 @@ def leaveEvent(self, event):
...
+		height = metrics.height()
+		self.resize(width, height)
+		leftSide = self.editor.lineNumberAreaWidth()
+		self.move(leftSide, viewport.height() - height)
+		self.setVisible(not globalSettings.useFakeVim)
+
+	def getText(self):
+		if not globalSettings.documentStatsEnabled:
+			return
+		template = '%d w : %d a : %d c'
```
Comment (mitya57) [inline ReText/editor.py]: Please make this translatable.

File: ReText/editor.py (hunk 2 above)
Comment (xgouchet) [inline ReText/editor.py]: Yes they were left over, sorry. 
On my laptop, for a 100k document (100'000 characters, 3500 lines), I get an average of 40ms to compute the stats

File: ReText/editor.py (hunk 3 above)
Comment (xgouchet) [inline ReText/editor.py]: :heavy_check_mark:

File: ReText/editor.py (hunk 1 above)
Comment (xgouchet) [inline ReText/editor.py]: :heavy_check_mark:

**Comment by xgouchet** [comment]
I updated the PR with your recommendation : a base class for additional information (LineInfo / Text Info), and updated the text analysis to correct the word count

**Comment by mitya57** [comment]
Thank you!

**Pull request merged**